mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/ReplAwareSparkDataSourceListener.scala

```diff
@@ -44,7 +44,18 @@ class ReplAwareSparkDataSourceListener(
     executionIdToReplId.put(executionId, replIdOpt.get)
   }
 
-  override protected def getReplIdOpt(event: SparkListenerSQLExecutionEnd): Option[String] = {
+  protected[autologging] override def onSQLExecutionEnd(event: SparkListenerSQLExecutionEnd): Unit = {
+    val extractor = getDatasourceAttributeExtractor
+    val tableInfos = extractor.getTableInfos(event)
+    val replIdOpt = popReplIdOpt(event)
+    if (replIdOpt.isDefined) {
+      tableInfos.foreach { tableInfo =>
+        publisher.publishEvent(replIdOpt = replIdOpt, sparkTableInfo = tableInfo)
```

Comment:
Fetch the REPL ID once and, only if it is defined, publish table read events to the subscriber(s) associated with the REPL